### PR TITLE
disttask: merge `GlobalTaskManager` and `SubTaskManager`

### LINKED ISSUE #42785: Merge `GlobalTaskManager` and `SubtaskManager`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #42785

### What is changed and how it works?

Merge `GlobalTaskManager` and `SubTaskManager` to `TaskManager` to manage both global and sub tasks. One task manager will be easier to do other future works like create subtask and global task in one transaction

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -327,25 +296,25 @@ func (stm *SubTaskManager) HasSubtasksInStates(tidbID string, taskID int64, stat
...
 	defer stm.mu.Unlock()
 
 	_, err := execSQL(stm.ctx, stm.se, "update mysql.tidb_background_subtask set exec_expired = %? where exec_id = %? and task_key = %?", heartbeat.String(), instanceID, taskID)
 	return err
 }
 
-// DeleteTasks deletes the subtask of the given global task ID.
-func (stm *SubTaskManager) DeleteTasks(taskID int64) error {
+// DeleteSubtasksUnderGlobalTask deletes the subtask of the given global task ID.
+func (stm *TaskManager) DeleteSubtasksUnderGlobalTask(taskID int64) error {
```
Comment (zimulala) [inline disttask/framework/storage/task_table.go:317]: `DeleteSubtasksUnderGlobalTask` rename it like `DeleteSubtasksByTaskID` or `DeleteSubtasksUnderTask` ?

File: disttask/framework/storage/task_table.go [hunk 2]
```diff
@@ -357,8 +326,8 @@ func (stm *SubTaskManager) DeleteTasks(taskID int64) error {
 	return nil
 }
 
-// GetSchedulerIDs gets the scheduler IDs of the given global task ID.
-func (stm *SubTaskManager) GetSchedulerIDs(taskID int64) ([]string, error) {
+// GetSchedulerIDsUnderGlobalTask gets the scheduler IDs of the given global task ID.
+func (stm *TaskManager) GetSchedulerIDsUnderGlobalTask(taskID int64) ([]string, error) {
```
Comment (zimulala) [inline disttask/framework/storage/task_table.go:330]: Ditto.

**Review by wjhuang2016 — APPROVED** [review]

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**